### ISSUE TITLE
change to generate telegraf.conf at build time breaks make packages for other archs than what you are running on

### ISSUE THREAD

**Issue by larsf** [influxdata/telegraf#13694]
### Relevant telegraf.conf

```toml
N/A
```

### Logs from Telegraf

```text
N/A
```

### System info

telegraf 1.27, debian:10 docker image

### Docker

N/A

### Steps to reproduce

Set GOARCH=<something other than the system you are building on>" like arm64 when building on an intel based linux node
GOARCH=arm64
make package

### Expected behavior

Successful built package for non-native arch

### Actual behavior

Fails because it tries to exec the newly build telegraf for the non-native arch to generate config

generating default config
go run ./cmd/telegraf config > etc/telegraf.conf
fork/exec /tmp/go-build708861788/b001/exe/telegraf: exec format error
make: *** [Makefile:125: config] Error 1

### Additional info

Caused by this commit;
chore: Generate config at build time (https://github.com/influxdata/telegraf/pull/12766)

**Comment by powersj** [comment]
> exec format error

This error means you are trying to run a binary that was not built on your architecture. Your steps to reproduce include setting `GOARCH`, and then running `go run`, which will not.

> Successful built package for non-native arch

You can cross-build the package just fine:

```s
❯ GOARCH=arm64 make telegraf
CGO_ENABLED=0 go build -tags "" -ldflags " -X github.com/influxdata/telegraf/internal.Commit=12490803 -X github.com/influxdata/telegraf/internal.Branch=ear/4392 -X github.com/influxdata/telegraf/internal.Version=1.28.0-12490803" ./cmd/telegraf
telegraf on  ear/4392 [?] via 🐹 v1.20.6 
❯ file telegraf 
telegraf: ELF 64-bit LSB executable, ARM aarch64, version 1 (SYSV), statically linked, Go BuildID=sNm5WIREFGxIgA70zanw/Tj8_DoyLj0Wnt0q1CUXV/l6K-uafq2_A1ZnSItlnv/VAOSwj-Gf8nXZwY8MSQ1, with debug_info, not stripped
```

However, you cannot generate the config for that package because it is a binary not meant for your architecture in the first place.

Previously, before #12766, we kept a version of the config in tree. However, this config was always wrong. It was either missing plugins, did not include plugins for certain platforms or architectures. As such we said enough of that.

Your options then are the following:

1. Use the config included in the packaging we provide from a nightly build or the releases page
2. Run the custom binary on the platform it was built for and generate the config
3. Build the binary for the platform you are actually on, generate the config, but know that there might be a slight difference.

**Comment by larsf** [comment]
Correct, but before this change, I was able to build a package for intel and arm64 on the same host successfully, so this change is a regression in that regard. In order to use make package and use one of the workaround you point out above, we need to add an option to the Makefile to make it skip trying to generate the telegraf.conf when you do make package

**Comment by powersj** [comment]
You are more than welcome to put up a PR. The makefile's packaging is not something I consider a regression similar to breaking a user's config or a bug introduced to a plugin.

**Comment by Hipska** [comment]
@larsf do you agree then to create a package for other architecture but missing a correct config file?
